Firebase/Auth/Source/FIRAuth.m

```diff
@@ -1151,6 +1177,38 @@ - (void)phoneNumberSignInWithRequest:(FIRVerifyPhoneNumberRequest *)request
 }
 #endif
 
+/** @fn internalSignInWithEmail:password:callback:
+    @brief Signs in using an email address and password.
+    @param email The user's email address.
+    @param password The user's password.
+    @param completion A block which is invoked when the sign in finishes (or is cancelled.) Invoked
+        asynchronously on the global auth work queue in the future.
+    @remarks This is the internal counterpart of this method, which uses a callback that does not
+        update the current user.
+ */
+- (void)internalSignInWithEmail:(NSString *)email
+                       password:(NSString *)password
+                     completion:(FIRVerifyPasswordResponseCallback)completion {
+  FIRVerifyPasswordRequest *request =
+      [[FIRVerifyPasswordRequest alloc] initWithEmail:email
+                                             password:password
+                                 requestConfiguration:_requestConfiguration];
+
+  if (![request.password length]) {
+    completion(nil, [FIRAuthErrorUtils wrongPasswordErrorWithMessage:nil]);
+    return;
+  }
+  [FIRAuthBackend verifyPassword:request
+                        callback:^(FIRVerifyPasswordResponse *_Nullable response,
+                                   NSError *_Nullable error) {
+    if (error) {
+      completion(nil, error);
+      return;
+    }
+    completion(response, error);
```

Comment:
Pass completion to `FIRAuthBackend verifiyPassword:callback:` directly.

Reply:
done, thanks